pilot/pkg/model/telemetry.go

```diff
@@ -169,3 +200,39 @@ func shallowMergeAccessLogs(parent *tpb.Telemetry, child *tpb.Telemetry) {
 		mergedLogging.Disabled = childLogging.Disabled
 	}
 }
+
+func TelemetryEnabled(spec *tpb.Telemetry, mesh *meshconfig.MeshConfig) bool {
+	return len(TelemetryProviders(spec, mesh)) > 0
+}
+
+func PrometheusEnabled(providers []*meshconfig.MeshConfig_ExtensionProvider) bool {
+	for _, prov := range providers {
+		if _, ok := prov.Provider.(*meshconfig.MeshConfig_ExtensionProvider_Prometheus); ok {
+			return true
+		}
+	}
+	return false
+}
+
+func TelemetryProviders(spec *tpb.Telemetry, mesh *meshconfig.MeshConfig) []*meshconfig.MeshConfig_ExtensionProvider {
```

Comment:
nit: here and above, should we add comments on the exported methods? I ask here, because I feel like this is more of a `MetricsProviders` at the moment (is that right?).

Reply:
ack, will do